R/cf.R

```diff
@@ -1024,6 +1024,106 @@ mul.cf <- function(cf, a=1.) {
   return (cf)
 }
 
+#' Exponentiate cf objects
+#'
+#' Note that no complex arithmetic is used, real and imaginary parts are 
+#' treated as seperate and indepenent, such that the real part of cf is
+#' raised to the power of n and similarly for the imaginary part.
+#'
+#' Note that this is generally only allowed on bootstrap samples and mean values,
+#' although it makes sense in some exeptional circumstances. Don't use this
+#' function unless you're certain that you should!
+#' 
+#' @param cf `cf_orig` object.
+#' @param n Numeric.
+#'
+#' @return
+#' The value is
+#' \deqn{cf^n \,.}
+#' @export
+pow.cf <- function(cf, n=1.) {
+  stopifnot(inherits(cf, 'cf_meta'))
+  if(inherits(cf, 'cf_orig')) {
+    cf$cf <- (cf$cf)^n
+    
+    if( has_icf(cf) ){
+      stopifnot(has_icf(cf))
```

Comment:
i don't think you need this check here, you've just checked it

Reply:
Yes, because I already checked with "if(inherits(cf, 'cf_orig'))" if the cf object inherits from the class "cf_orig" which contains icf.